keps/sig-scheduling/4328-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,359 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP aims to introduce affinity-based eviction.
+In this KEP, we will introduce a new nodeAffinity type: `requiredDuringSchedulingRequiredDuringExecution`.
+A new controller called `affinity-eviction` will be added to the controller-manager. 
+This controller will monitor changes in node labels and evict pods when their strict node affinity is no longer met.
+
+## Motivation
+
+- Enable strict node affinity for pod scheduling and execution.
+- Provide users with greater control over pod placement.
+- Ensure predictable and consistent pod behavior in various scenarios.
+- Address use cases with specific placement and execution requirements.
+  
+### Goals
+
+- Introduce the `RequiredDuringSchedulingRequiredDuringExecution` **nodeAffinity** type.
+- Add `affinityEviction` controller to ensure pods being evicted if the selectors are no longer met at runtime.
+
+### Non-Goals
+
+- This KEP does not aim to Introduce the `RequiredDuringSchedulingRequiredDuringExecution` **podAffinity** type. 
+- This KEP does not aim to Introduce the `RequiredDuringSchedulingRequiredDuringExecution` **podAntiAffinity** type. 
+  
+## Proposal
+- Implement the "RequiredDuringSchedulingRequiredDuringExecution" node affinity type.
+- Update the kube-scheduler, kube-controller-manager and kube-apiserver.
+- Define the behavior, syntax, and use cases in documentation.
+- Facilitate testing, adoption, and maintenance of this feature.
+
+### User Stories (Optional)
+
+#### Optimizing Workloads Migrating
+As a Data Scientist, I want to ensure that my GPU-intensive ml workloads are always on nodes with correct type of GPUs during both scheduling and execution. 
+So I labeled one node with "gpu=g1". I added some more powerful nodes, and now I want to move all pods requiring gpu to this new node. 
+With this feature, I can simply remove the "gpu=g1" from the first node and adds it to the new node, so I can move the target pods and keep the remaining pods intact.
+So far descheduler has the ability to achieve this goal. However, bringing a new component to the system introduces both a lot of uncertainty and additional learning costs. So as a user, I hope that kubernetes can natively support this feature so that users can do it without spending extra learning costs.
```

Comment:
```suggestion
So far descheduler has the ability to achieve this goal. 
However, bringing a new component to the system introduces both a lot of uncertainty and additional learning costs. 
So as a user, I hope that kubernetes can natively support this feature so that users can do it without spending extra learning costs.
```
Please try to have new lines between sentences. Nit